acceptance/tests/pxp-module-puppet/run_puppet_result_failed.rb

```diff
@@ -0,0 +1,94 @@
+    end
+  end
+
+  target_identities = []
+  step "Send an rpc_blocking_request to all agents for #{tmp_environment}" do
+    agents.each do |agent|
+      target_identities << "pcp://#{agent}/agent"
+    end
+    responses = nil # Declare here so not local to begin/rescue below
+    begin
+      responses = rpc_blocking_request(master, target_identities,
+                                      'pxp-module-puppet', 'run',
+                                      {:env => [], :flags => ['--environment', tmp_environment,
+                                                              '--onetime']
+                                      })
+    rescue => exception
+      fail("Exception occurred when trying to run Puppet on all agents: #{exception.message}")
+    end
+    agents.each do |agent|
+      step "Check Run Puppet response for #{agent}" do
+        identity = "pcp://#{agent}/agent"
+        action_result = responses[identity][:data]["results"]
+        assert(action_result.has_key?('status'),
+               "Results for pxp-module-puppet run on #{agent} should contain a 'status' field")
+        assert_equal(tmp_environment, action_result['environment'],
+                     "Result of pxp-module-puppet run on #{agent} should run with the "\
+                     "#{tmp_environment} environment")
+        assert_equal('failed', action_result['status'],
+                     "Result of pxp-module-puppet run on #{agent} should be 'failed'. was: "\
+                     "'#{action_result["status"]}'")
+      end
+    end
+  end
+
+  step "Send an rpc_blocking_request to all agents for #{tmp_environment2}" do
+    responses = nil # Declare here so not local to begin/rescue below
+    begin
+      responses = rpc_blocking_request(master, target_identities,
+                                      'pxp-module-puppet', 'run',
+                                      {:env => [], :flags => ['--environment', tmp_environment2,
+                                                              '--onetime']
+                                      })
+    rescue => exception
+      fail("Exception occurred when trying to run Puppet on all agents: #{exception.message}")
+    end
+    agents.each do |agent|
+      step "Check Run Puppet response for #{agent}" do
+        identity = "pcp://#{agent}/agent"
+        action_result = responses[identity][:data]["results"]
+        assert(action_result.has_key?('status'),
+               "Results for pxp-module-puppet run on #{agent} should contain a 'status' field")
+        assert_equal(tmp_environment2, action_result['environment'],
+                     "Result of pxp-module-puppet run on #{agent} should run with the "\
+                     "#{tmp_environment2} environment")
+        assert_equal('failed', action_result['status'],
+                     "Result of pxp-module-puppet run on #{agent} should be 'failed'. was: "\
```

Comment:
I  think if the comment above is addressed; the result JSON will contain both a PXP action result and Puppet agent results, with the action result being "success" and the Puppet agent result being "failed" - this test should probably assert both